### PR TITLE
fix: prevent body parser error

### PR DESCRIPTION
**Pull request by SuperOleg39**
After integrating fastify, POST requests to my application stopped working.
The problem is somewhere in the depths of `body-parser` library for express. If parsing for body from this library works, request hangs, and fastify when calling next in `onRequest` doesn't reach routing, request just hangs.

We're not the only ones who are like this - https://github.com/fastify/fastify-express/issues/73#issuecomment-1067022626

To solve this problem, we can reject `body-parser` and `cookie-parser` as well.
To maintain backward compatibility, we can use already prepared parsing result from `fastify-cookie` and `fastify-formbody`, put it into `req.raw.cookies` and `req.raw.body` respectively.
Then we need to move the express call to the `preHandler` hook, since we don't have the parsing results ready before that.

### CODE REVIEW

**Comment by SuperOleg39** [comment]
`res.end should block middleware execution` test case are broken with this changes.
Is this case are really important?

File: .history/index_20220427135333.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+'use strict'
```
[the remark this answers is not in the document]
Comment (SuperOleg39) [reply .history/index_20220427135333.js]: vscode history plugin, my mistake, thx

File: index.js [hunk 2]
```diff
@@ -12,8 +12,8 @@ function expressPlugin (fastify, options, next) {
   fastify.express.disable('x-powered-by')
 
   fastify
-    .addHook('onRequest', enhanceRequest)
-    .addHook('onRequest', runConnect)
+    .addHook('preHandler', enhanceRequest)
+    .addHook('preHandler', runConnect)
```
Comment (mcollina) [inline index.js:16]: Could you make the hook configurable via an option, defaulting to `onRequest`?

**Review by mcollina — CHANGES_REQUESTED** [review]
Thanks and good spot! I'm
not sure what is the setup for your express app, however how things are wired here have worked for some people (including myself).

Could you make the hook configurable, document the new option in the README, and add some tests? Thanks!

File: index.js (hunk 2 above)
Comment (SuperOleg39) [reply index.js]: Done. But not the best naming, I think..

**Comment by SuperOleg39** [comment]
> Thanks and good spot! I'm not sure what is the setup for your express app, however how things are wired here have worked for some people (including myself).
> 
> Could you make the hook configurable, document the new option in the README, and add some tests? Thanks!

Yes, add tests and troubleshooting section)

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by SuperOleg39** [comment]
Looks like I added unstable tests?

**Comment by mcollina** [comment]
Doesn't look like it, things are passing on master

**Comment by SuperOleg39** [comment]
Nice, thanks!

**Pull request merged**